src/API/index.js

```diff
@@ -1,87 +1,91 @@
+	  let config = {
+		  method: 'PUT',
+		  credentials: "include",
+		  body: newData,
+		  headers: {
+			  Authorization: `Bearer ${token}`,
+		  },
+	  }
+	  let response = await fetch(base_url + `document/${question_id}/data`, config);
+	  return response.json();
+  },
+
+  async createQuestion(question, token) {
+    return baseRequest(`document`, 'POST', question, token);
+  },
+  async updateQuestion(question, token) {
+    return baseRequest(`document/${question.id}`, 'PUT', question, token);
+  },
+  async deleteQuestion(question_id, token) {
+    return baseRequest(`document/${question_id}`, 'DELETE', newQuestion, token);
+  },
 
-function errorHandling(err) {
-  /* eslint-disable no-console */
-  let errorText = '';
-  if (err.response) {
-    console.log('Error Status', err.response.status);
-    console.log('Error Downloading', err.response.data);
-    errorText = err.response.data;
-  } else if (err.request) {
-    console.log('No response was received', err.request);
-    errorText = 'There was no response from the server.';
-  } else {
-    console.log('Unknown Error', err.message);
-    errorText = err.message;
-  }
-  return errorText;
-  /* eslint-enable no-console */
 }
 
-const routes = {
-  getUser: () => new Promise((resolve, reject) => {
-    axios.get('/api/user')
-      .then((res) => {
-        console.log('user res', res);
-        resolve(res.data);
-      })
-      .catch((err) => {
-        const errorText = errorHandling(err);
-        reject(errorText);
-      });
-  }),
-  setUser: (token, username) => axios.post('/api/user', { token, username }),
-  getAllQuestions: () => new Promise((resolve, reject) => {
-    axios.get('/api/questions')
-      .then((res) => {
-        console.log('Questions', res);
-        resolve(res);
-      })
-      .catch((err) => {
-        const errorText = errorHandling(err);
-        reject(errorText);
-      });
-  }),
-  createQuestion: (question, token) => new Promise((resolve, reject) => {
-    const config = {
-      method: 'get',
-      url: '/api/questions',
-      headers: {},
-    };
-    if (token) {
-      config.headers.authorization = `Bearer ${token}`;
-    }
-    axios.request({
-      method: 'post',
-      url: '/api/questions',
-      data: question,
-    })
-      .then((res) => {
-        resolve(res);
-      })
-      .catch((err) => {
-        const errorText = errorHandling(err);
-        reject(errorText);
-      });
-  }),
-  getQuestions: (token) => new Promise((resolve, reject) => {
-    const config = {
-      method: 'get',
-      url: '/api/questions',
-      headers: {},
-    };
-    if (token) {
-      config.headers.authorization = `Bearer ${token}`;
-    }
-    axios.request(config)
-      .then((res) => {
-        resolve(res);
-      })
-      .catch((err) => {
-        const errorText = errorHandling(err);
-        reject(errorText);
-      });
-  }),
-};
+// Some of the API routes have the same method signatures for questions and answers.
```

Comment:
If we're going to reuse these methods for both answers and questions, I would rather give the methods generic names like getDocument, createDocument, because we'll be exposing better method names down below. Then we won't have the getAnswer: routes.getQuestion, which is weird at a glance. You would need to update the exposed UI methods that are currently in the spread operator.